releases/release-5.1.2.md

```diff
@@ -0,0 +1,156 @@
+    - Fix a bug 'select for update' does not correctly lock the data when a normal table join a partition table. [#26631](https://github.com/pingcap/tidb/pull/26631)
+    - Fix a panic when select for update on a normal table join a partition table [#26563](https://github.com/pingcap/tidb/pull/26563)
+    - Fix the issue that point get does not use lite version resolve lock [#26562](https://github.com/pingcap/tidb/pull/26562)
+
++ TiKV
+
+    - Fix bug when TiKV upgrade from 3.0 to 4.x and 5.x but there are still some files left by import. [#10912](https://github.com/tikv/tikv/pull/10912)
+    - RaftStore Snapshot GC fix: fix the issue that snapshot GC missed GC snapshot files when there's one snapshot file failed to be GC-ed. [#10873](https://github.com/tikv/tikv/pull/10873)
+    - TiKV coprocessor slow log will only consider time spent on processing the request.
+    - Drop log instead of blocking threads when slogger thread is overloaded and queue is filled up. [#10865](https://github.com/tikv/tikv/pull/10865)
+    - Bug fix: fix an unexpected panic when exceeds deadline on processing copr requests. [#10856](https://github.com/tikv/tikv/pull/10856)
+    - Fix TiKV panic when enable Titan and upgrade from pre-5.0 version
+    - Fix newer TiKV can't rollback to 5.0.x [#10842](https://github.com/tikv/tikv/pull/10842)
+    - Fix TiKV delete files before it ingests to RocksDB. [#10741](https://github.com/tikv/tikv/pull/10741)
+    - Support changing CDC configs dynamically [#10686](https://github.com/tikv/tikv/pull/10686)
+    - Reduce resolved ts message size to save network bandwidth. [#10679](https://github.com/tikv/tikv/pull/10679)
+    - Fix the resolve failures caused by the left pessimisic locks. [#10653](https://github.com/tikv/tikv/pull/10653)
+    - limit the hotspot report count. [#10621](https://github.com/tikv/tikv/pull/10621)
+
++ PingCAP/TiFlash
+
+    - function result name should contain collator info [#2819](https://github.com/pingcap/tics/pull/2819)
+    - Support INET6_ATON and INET6_NTOA in TiFlash. [#2624](https://github.com/pingcap/tics/pull/2624)
+    - Support INET_ATON and INET_NTOA in TiFlash. [#2613](https://github.com/pingcap/tics/pull/2613)
+    - update client to fix race bug [#2583](https://github.com/pingcap/tics/pull/2583)
+    - expand streams after aggregation [#2537](https://github.com/pingcap/tics/pull/2537)
+    - Improve the mpp cancel process to cancel the mpp task ASAP [#2518](https://github.com/pingcap/tics/pull/2518)
+    - Fix the bug may cause metrics to display error value. [#2469](https://github.com/pingcap/tics/pull/2469)
+
++ PD
+
+    - allow empty region to be scheduled and use a sperate tolerance config in scatter range scheduler [#4117](https://github.com/tikv/pd/pull/4117)
+    - Improved the performance of synchronizing Region information between PDs. [#3933](https://github.com/tikv/pd/pull/3933)
+
```

Comment:
- dynamically adjust the retry limit according to the operator [#4048](https://github.com/tikv/pd/pull/4048)
-